server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconciler.java

```diff
@@ -100,33 +108,29 @@ public DesiredBalanceReconciler(ClusterSettings clusterSettings, ThreadPool thre
             value -> this.undesiredAllocationsLogThreshold = value
         );
 
-        unassignedShards = LongGaugeMetric.create(
-            meterRegistry,
+        meterRegistry.registerLongsGauge(
             "es.allocator.desired_balance.shards.unassigned.current",
             "Current number of unassigned shards",
-            "{shard}"
+            "{shard}",
+            this::getUnassignedShards
         );
-        totalAllocations = LongGaugeMetric.create(
-            meterRegistry,
+        meterRegistry.registerLongsGauge(
             "es.allocator.desired_balance.shards.current",
             "Total number of shards",
-            "{shard}"
+            "{shard}",
+            this::getTotalAllocations
         );
-        undesiredAllocations = LongGaugeMetric.create(
-            meterRegistry,
+        meterRegistry.registerLongsGauge(
```

Comment:
I am new to this one.
Does it emit a single document with array of metrics or number of separate documents?
In other words, if not an elected master, would we get empty array or no documents?

Reply:
The places I've seen it used is where you have a bunch of metrics that have different labels. So a number of separate documents. In this case we're just using it to be able to publish nothing when we are not the master (i.e. no documents)